src/main/java/com/tiff/tffnserachservice/controller/TiffnerController.java

```diff
@@ -0,0 +1,64 @@
+package com.tiff.tffnserachservice.controller;
+
+
+import com.tiff.tffnserachservice.dto.TiffnerDTO;
+import com.tiff.tffnserachservice.model.Tiffner;
+import com.tiff.tffnserachservice.repository.TiffnerRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/tffn-search")
+public class TiffnerController {
+
+    @Autowired
+    private TiffnerRepository tiffnerRepository;
+
+    @GetMapping
+    public List<Tiffner> findTiffns() {
+        return tiffnerRepository.findAll();
+    }
+
+    @GetMapping("/api/find-tiffn/{id}")
+    public ResponseEntity<Tiffner> findTffn(@PathVariable Long id) {
+        return tiffnerRepository.findByTiffnerId(id).map(ResponseEntity::ok).orElseThrow();
+    }
+
+
+    @DeleteMapping("/api/delete-tiffn/{id")
+    public Boolean deleteTffn(Long id) {
+
+        if (tiffnerRepository.findByTiffnerId(id).isPresent()) {
+            tiffnerRepository.delete(tiffnerRepository.findByTiffnerId(id).orElseThrow());
+            return true;
+        }
+        return false;
+    }
+
+    @PostMapping()
+    public void addTffn(Tiffner tiffner){
+         tiffnerRepository.save(tiffner);
+    }
+
+    @PutMapping("/api/update-tiffn/{id}")
+    public ResponseEntity<Tiffner> updateTffn(@RequestBody TiffnerDTO tiffnerDTO, Long id){
+        Tiffner tiffner = tiffnerRepository.findByTiffnerId(id).orElseThrow();
+
+        tiffner.setName(tiffnerDTO.getName());
```

Comment:
We should use the dto to transfer the properties the user provides to use in the request body to a given tffn object, we should never go about manually setting the properties/fields. Think about what happens when we add an ew field, and each time we have to come here and add that field and set it, thats too much of a constraint. Instead utilize the BeansUtil method the purpose of DTOs are in the name Data Transfer Object, they are used to transfer one objects data into another, the only time we manually want to override values here is if there is some compute that needs to be happen for a given value before we can set it in the tffn object.